src/types/response/guards.ts

```diff
@@ -23,3 +26,34 @@ export const isProcessedSingleAnswerResponse = (
     isProcessedFieldResponse(response)
   )
 }
+
+export const isProcessedCheckboxResponse = (
+  response: any,
+): response is ProcessedCheckboxResponse => {
+  return (
+    'answerArray' in response &&
+    Array.isArray(response.answerArray) &&
+    isProcessedFieldResponse(response)
+  )
+}
+
+export const isProcessedTableResponse = (
+  response: any,
+): response is ProcessedTableResponse => {
+  return (
+    'answerArray' in response &&
+    Array.isArray(response.answerArray) &&
+    Array.isArray(response.answerArray[0]) &&
```

Comment:
this check looks weird - should every element in `answerArray` be an array? and should every element in it be of string type?

Reply:
there are important ramifications for the type safety of downstream code, if the type guard is not written as strictly as possible

Reply:
you may also want to introduce new subtypes with additional type guards for each row of answers, and build your way up to having a bigger `isProcessedTableResponse`. i suspect the subtypes will come in useful for composing table row validators

Reply:
thanks! see if the edits work